### DISCUSSION TITLE
BRouter websites / Cruiser: different routing results!

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#574] [Q&A] [answered]
When planning with the BRouter WebClient, completely different routes come out than when planning with the Cruiser!
Every time with the same route profile.
????

**Comment by devemux86** [comment]
Can you post a specific route example?
Which website, the official or another one?

What options do you use in Cruiser's _advanced_ BRouter settings?

[BRouter website](https://brouter.de/brouter-web/) probably has not updated the BRouter library (latest BRouter is 1.7.5).
If we export a route from BRouter website in GeoJSON, it has as creator "BRouter-1.7.0".

[bikerouter.de](https://bikerouter.de/) mentions creator "BRouter-1.7.4".

**Reply by Leeb64** [reply]
The planning was done with my country road profile 3+4 category. At Original BRouter web.

[Landstraße  3+4 .txt](https://github.com/user-attachments/files/15911882/Landstrasse.3%2B4.txt)

![grafik](https://github.com/devemux86/cruiser/assets/120722053/c46273ee-c37b-40ac-af2d-2b5228eb8bc7)
mit Web Client
[BRouter webclient (721,9 km).geojson.txt](https://github.com/user-attachments/files/15911947/BRouter.webclient.721.9.km.geojson.txt)


![20240620_113321](https://github.com/devemux86/cruiser/assets/120722053/acec43ec-9963-4c00-8493-8e34fd0cf144)
mit Cruiser
[Cruiser.json.txt](https://github.com/user-attachments/files/15911958/Cruiser.json.txt)

**Reply by Leeb64** [reply]
The result is better with BRouter Web (I think).
You could still bend that...
But the problem is more the huge difference.

**Reply by devemux86** [reply]
[BRouter website](https://brouter.de/brouter-web/) has more old library code and we do not know how the city / forest options work there.
You can try without the city / forest options.

Also try the [bikerouter.de](https://bikerouter.de/t) that has more recent BRouter code and produces a route similar with Cruiser.

Some users have also mentioned some other BRouter websites.

**Reply by Leeb64** [reply]
Apparently everyone does what they want...
It's similar with the bike router... from a distance. But it doesn't avoid driving through town like in the cruiser... so it's no good either.
somehow not so exciting!
By the way, this bike router is extremely lame...you can just have a quick coffee there.

**Reply by devemux86** [reply, marked as answer]
The problem is the town + forest options at the same time:
- If we enable only the _avoid towns_, the routes are similar.
- If we enable them both, the routes are not similar.
- If we disable them, the routes are similar.

Anyway, it's always recommended using the avoid towns and not all options together, as they fight each other.
Perhaps their rules in the profile need improvements so they can work better together. :slightly_smiling_face:

**Reply by devemux86** [reply]
Similar routes with the avoid towns option:

<img src="https://github.com/devemux86/cruiser/assets/3484020/300677f7-c410-4ee6-a459-9c43a1316a73" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/c469db2b-59f7-4ce4-b20b-b63254d46c35" width="50%"/>

**Reply by Leeb64** [reply]
I still can't fully understand it...
If all the settings are the same for the same program, then the result should actually be the same.
Anything else would be magic...or so.

**Reply by devemux86** [reply]
BRouter websites use different BRouter versions, so their routes can be different, especially with new options.
App|BRouter
-|-
Cruiser|BRouter latest
https://brouter.de/brouter-web/|BRouter 1.7.0
https://bikerouter.de/|BRouter 1.7.4
https://brouter.de/essbee/ | BRouter 1.7.0

**Reply by devemux86** [reply]
There is also https://brouter.de/essbee/ with BRouter 1.7.0 by @essbee who worked on the BRouter pseudo-tags.

Documentation about the BRouter pseudo-tags: https://brouter.de/essbee/infoPseudoTags.html

**Comment by devemux86** [comment]
Important:

_Different route profiles for sections are provided only inside Cruiser._ :slightly_smiling_face: 

BRouter websites calculate the whole route with the same profile.
